### ISSUE TITLE
Bug: ErrorException: array_key_exists(): Using array_key_exists() on objects is deprecated. Use isset() or property_exists() instead from SYSTEMPATH\HTTP\Files\FileCollection.php at line 120

### ISSUE THREAD

**Issue by hurtsky** [codeigniter4/CodeIgniter4#6255]
### PHP Version

7.4

### CodeIgniter4 Version

develop : ddccfc07e1ee06edf2652d5ef13e306496c83365

### CodeIgniter4 Installation Method

Git

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

5.6

### What happened?

uploading a file while using below code , will thrown `ErrorException: array_key_exists(): Using array_key_exists() on objects is deprecated. Use isset() or property_exists() instead`.

attached screenshot: 
![screencapture-127-0-0-1-sstms-backend-public-user-upload-photo-2022-07-12-18_20_19](https://user-images.githubusercontent.com/4039059/178468985-f360df19-946b-4ef6-abca-315c2537393b.png)

### Steps to Reproduce

```
    public function submit_photo()
    {
        if (!$this->validate([
            'upload_file_photo' => [
                'label' => 'Upload Photo',
                'rules' => 'uploaded[upload_file_photo]'
                    . '|is_image[upload_file_photo]'
                    . '|mime_in[upload_file_photo,image/jpg,image/jpeg]'
                    . '|max_size[upload_file_photo.2048]',
            ],
        ])) {
            return redirect()->back()->with('validations', $this->validator->getErrors());
        }
        $path = WRITEPATH . 'uploads/user/' . session()->user_id . '/img';
        $file = $this->request->getFile('upload_file_photo');
        if ($file->move($path, 'profile.jpg', true)) {
            return redirect()->back()->with(
                'responses',
                'success'
            );
        }

        return redirect()->back()->with(
            'responses',
            'failed'
        );
    }
```

### Expected Output

Should not throw ErrorException

**Comment by hurtsky** [comment]
looks like the use of max_size rule trigger the ErrorException, removing it seems okay

**Comment by kenjis** [comment]
You set `max_size[upload_file_photo.2048]`, but the usage of `max_size` is like `max_size[field_name,2048]`.

**Comment by hurtsky** [comment]
> You set `max_size[upload_file_photo.2048]`, but the usage of `max_size` is like `max_size[field_name,2048]`.

Thank you!

**Comment by kenjis** [comment]
This error message is not friendly, and it is a bug.
I sent a PR to fix it: #6261

**Comment by hurtsky** [comment]
might related to this, I just did some testing with  mime_in rules, if the developer accidentally put the `,` to `.` it will throw same error. I'm not checking all the file validation rules, but might be the others rules also have this kind of error message
